spec/services/atmosphere/billing_service_spec.rb

```diff
@@ -148,6 +174,11 @@
       expensive_vm = create(:virtual_machine, compute_site: cs, source_template: tmpl_of_not_shareable_at, virtual_machine_flavor: expensive_flavor)
       cheap_vm = create(:virtual_machine, compute_site: cs, source_template: tmpl_of_not_shareable_at, virtual_machine_flavor: cheap_flavor)
       free_vm = create(:virtual_machine, compute_site: cs, source_template: tmpl_of_not_shareable_at, virtual_machine_flavor: free_flavor)
+      shareable_vm = create(:virtual_machine, compute_site: cs, source_template: tmpl_of_shareable_at, virtual_machine_flavor: free_flavor)
+
+      free_flavor.virtual_machine_flavor_os_families.first.update_attribute(:hourly_cost, 0)
+      cheap_flavor.virtual_machine_flavor_os_families.first.update_attribute(:hourly_cost, 10)
+      expensive_flavor.virtual_machine_flavor_os_families.first.update_attribute(:hourly_cost, 100)
```

Comment:
Flavor definitions can be moved to above context and be reused. Consider testing only one aspect in one example.